core/mbed/source/interrupts.cpp

```diff
@@ -26,6 +26,11 @@ void vIRQ_SetVectorX(uint32_t irqn, uint32_t vector, uint32_t flag)
     }
 }
 
+void vIRQ_SetVector(uint32_t irqn, uint32_t vector)
```

Comment:
I didn't notice they have different signatures. It would be a dumb idea to break backwards compatibility just to avoid a function call in non-critical path code.

What is the reason why a call gateway for vIRQ_SetVector is needed? Could the gateway be just as effective at the vIRQ_SetVectorX call?

On 15 Dec 2015, at 21:17, Alessandro Angelino <notifications@github.com<mailto:notifications@github.com>> wrote:


In core/mbed/source/interrupts.cpp<https://github.com/ARMmbed/uvisor/pull/121#discussion_r47699617>:

> @@ -26,6 +26,11 @@ void vIRQ_SetVectorX(uint32_t irqn, uint32_t vector, uint32_t flag)
>      }
>  }
>
> +void vIRQ_SetVector(uint32_t irqn, uint32_t vector)


Yes that is an option! vIRQ_SetVector keeps backward-compatibility with the corresponding NVIC API, meaning that it uses the same format (irqn, vector). vIRQ_SetVectorX is somehow an extension, since it includes an additional flag -- although currently unused.

@meriac<https://github.com/meriac> what do you think? Are we willing to keep the flag of vIRQ_SetVectorX or do we prefer sticking with the classic API?

For reference:

void vIRQ_SetVector(uint32_t irqn, uint32_t vector);
void vIRQ_SetVectorX(uint32_t irqn, uint32_t vector, uint32_t flag);

-
Reply to this email directly or view it on GitHub<https://github.com/ARMmbed/uvisor/pull/121/files#r47699617>.

IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.